st2client/st2client/base.py

```diff
@@ -225,7 +225,7 @@ def _get_cached_auth_token(self, client, username, password):
         :rtype: ``str``
         """
         if not os.path.isdir(ST2_CONFIG_DIRECTORY):
-            os.makedirs(ST2_CONFIG_DIRECTORY)
+            os.makedirs(ST2_CONFIG_DIRECTORY, mode=0o770)
```

Comment:
Can we also set `setgid` for the dir? `2770`,

see https://github.com/StackStorm/st2/issues/4144 and https://github.com/StackStorm/packer-st2/issues/38 why

Reply:
Yes, I'll do this.

I assume we also want the `setuid` bit set, right? `0o6770`?

Reply:
I don't have any opinions about the `setuid`, since I didn't see that bit having effect on dir.
If you believe it can help in some use cases, - I'm good with adding it.

Reply:
Nope, I don't think it's helpful.

Fixed.